backend/models/bcf.js

```diff
@@ -408,816 +464,639 @@ function getBCFMarkup(issue, account, model, unit) {
+			issue.owner = _.get(xml, "Markup.Topic[0].CreationAuthor[0]._");
+			issue.extras.ModifiedDate = _.get(xml, "Markup.Topic[0].ModifiedDate[0]._");
+			issue.extras.ModifiedAuthor = _.get(xml, "Markup.Topic[0].ModifiedAuthor[0]._");
+			if (_.get(xml, "Markup.Topic[0].DueDate[0]._")) {
+				issue.due_date = moment(_.get(xml, "Markup.Topic[0].DueDate[0]._")).valueOf();
 			}
-
-			function createGroupData(groupObject) {
-
-				const groupData = {};
-
-				groupData.name = groupObject.name;
-				groupData.color = groupObject.color;
-
-				for (const groupAccount in groupObject.objects) {
-					for (const groupModel in groupObject.objects[groupAccount]) {
-						if (!groupData.objects) {
-							groupData.objects = [];
-						}
-
-						groupData.objects.push({
-							account: groupAccount,
-							model: groupModel,
-							ifc_guids: groupObject.objects[groupAccount][groupModel].ifc_guids
-						});
-					}
-				}
-
-				return groupData;
+			if (_.get(xml, "Markup.Topic[0].AssignedTo[0]._")) {
+				issue.assigned_roles = _.get(xml, "Markup.Topic[0].AssignedTo[0]._").split(",");
+			} else {
+				issue.assigned_roles = [];
 			}
 
-			function createGroupObject(group, name, color, groupAccount, groupModel, ifc_guid) {
+			issue.desc = (_.get(xml, "Markup.Topic[0].Description[0]._")) ? _.get(xml, "Markup.Topic[0].Description[0]._") : "(No Description)";
+			issue.extras.BimSnippet = _.get(xml, "Markup.Topic[0].BimSnippet");
+			issue.extras.DocumentReference = _.get(xml, "Markup.Topic[0].DocumentReference");
+			issue.extras.RelatedTopic = _.get(xml, "Markup.Topic[0].RelatedTopic");
+		}
 
-				if (groupAccount && groupModel && ifc_guid) {
-					if (!group) {
-						group = {};
-					}
+		_.get(xml, "Markup.Comment") && xml.Markup.Comment.forEach(comment => {
+			const obj = {
+				guid: _.get(comment, "@.Guid") ? utils.stringToUUID(_.get(comment, "@.Guid")) : utils.generateUUID(),
+				created: moment(_.get(comment, "Date[0]._")).format("x").valueOf(),
+				owner: _.get(comment, "Author[0]._"),
+				comment: _.get(comment, "Comment[0]._"),
+				sealed: true,
+				viewpoint: utils.isUUID(_.get(comment, "Viewpoint[0].@.Guid")) ? utils.stringToUUID(_.get(comment, "Viewpoint[0].@.Guid")) : undefined,
+				extras: {}
+			};
 
-					if (name) {
-						group.name = name;
-					}
+			obj.extras.ModifiedDate = _.get(comment, "ModifiedDate");
+			obj.extras.ModifiedAuthor = _.get(comment, "ModifiedAuthor");
 
-					if (color) {
-						group.color = color;
-					}
-
-					if (!group.objects) {
-						group.objects = {};
-					}
+			issue.comments.push(obj);
+		});
 
-					if (!group.objects[groupAccount]) {
-						group.objects[groupAccount] = {};
-					}
+		return {
+			issue,
+			viewpointsData: xml.Markup.Viewpoints
+		};
+	});
+}
 
-					if (!group.objects[groupAccount][groupModel]) {
-						group.objects[groupAccount][groupModel] = { ifc_guids: [] };
-					}
+bcf.readBCF = function(account, model, requester, ifcToModelMap, zipPath, settings) {
```

Comment:
this doesn't need exporting... it's a private function. don't include it inside bcf (and any other private functions)